habitat_sim/utils/viz_utils.py

```diff
@@ -26,6 +31,260 @@ def is_notebook():
+                res_image = cv2.resize(
+                    res_image, videodims, interpolation=cv2.INTER_AREA
+                )
+
+            # write the desired image to video
+            writer.append_data(res_image)
+    else:
+        for ob in observations:
+            res_image = cv2.cvtColor(ob[primary_pov], cv2.COLOR_RGBA2RGB)
+            if res_image.shape[:2] != videodims:
+                res_image = cv2.resize(
+                    res_image, videodims, interpolation=cv2.INTER_AREA
+                )
+            # write the desired image to video
+            writer.append_data(res_image)
+    writer.close()
+    if open_vid:
+        print("Displaying video")
+        display_video(video_file)
+
+
+def build_multi_obs_image(
+    base_camera_img, embed_image_data_list, thumb_size, outline_frame
+):
+    """
+    Build an image containing multiple sub-images
+
+    Parameters
+    ----------
+    base_camera_img : 2d numpy array
+        Base image to insert sub images into.
+    embed_image_data_list : list of tuples
+        List of tuples where 
+            idx 0 is numpy array of image data 
+            idx 1 is boolean whether is depth data or not.
+    thumb_size : tuple of ints
+        X and Y dimensions of inset images
+    outline_frame : numpy array
+        Max X and Y dims of frame surrounding inset image
+
+    Returns
+    -------
+    result_image : 2d numpy array
+        Resultant compound image
+
+    """
+    # Main Image
+    result_image = cv2.cvtColor(base_camera_img, cv2.COLOR_RGBA2RGB)
+
+    x_offset = 50
+    y_offset = 50
+    y_bound = 10
+    for embed_image, embed_image_is_depth in embed_image_data_list:
+
+        if embed_image_is_depth:
+            rgb_embed_image = clip_depth_image(embed_image)
+        else:
+            rgb_embed_image = cv2.cvtColor(embed_image, cv2.COLOR_RGBA2RGB)
+
+        resized_rgb_embed_image = cv2.resize(
+            rgb_embed_image, thumb_size, interpolation=cv2.INTER_AREA
+        )
+
+        result_image[
+            y_offset - 1 : y_offset + outline_frame.shape[0] - 1,
+            x_offset - 1 : x_offset + outline_frame.shape[1] - 1,
+        ] = outline_frame
+        result_image[
+            y_offset : y_offset + resized_rgb_embed_image.shape[0],
+            x_offset : x_offset + resized_rgb_embed_image.shape[1],
+        ] = resized_rgb_embed_image
+        # move along y axis
+        y_offset = y_offset + y_bound + thumb_size[1]
+
+    return result_image
+
+
+def clip_depth_image(depth_image, clip_max=10.0):
+    """
+    Manually normalize depth into [0, 1] so that images are always consistent
+
+    Parameters
+    ----------
+    depth_image : 2d numpy array
+        Depth image data
+
+    Returns
+    -------
+    rgb_d_im : 2d numpy array
+        Clipped depth image data
+
+    """
+    d_im = np.clip(depth_image, 0, clip_max)
+    d_im /= clip_max
+    rgb_d_im = cv2.cvtColor((d_im * 255).astype(np.uint8), cv2.COLOR_GRAY2RGB)
```

Comment:
This should be pure OpenCV or pure Numpy. Also you are combing the color conversion and RGB generation into the same function, but it's not obvious except for the return type.